src/data/finsupp.lean

```diff
@@ -1257,4 +1368,88 @@ protected def dom_congr [decidable_eq α₁] [decidable_eq α₂] [decidable_eq
     exact map_domain_id
   end⟩
 
+section sigma
+
+variables {αs : ι → Type*} [decidable_eq ι] [has_zero β] (l : (Σ i, αs i) →₀ β)
+
+noncomputable def split (i : ι) : αs i →₀ β :=
+l.comap_domain (sigma.mk i) (λ x1 x2 _ _ hx, heq_iff_eq.1 (sigma.mk.inj hx).2)
+
+lemma split_apply (i : ι) (x : αs i) : split l i x = l ⟨i, x⟩ :=
+begin
+ dunfold split,
+ rw comap_domain_apply
+end
+
+def split_support : finset ι := finset.image (sigma.fst) l.support
```

Comment:
Shouldn't this be a theorem? `(split l i).support = finset.image (sigma.fst) l.support`

Reply:
No, `(split l i).support` is of type `finset (αs i)`, but `finset.image (sigma.fst) l.support` is of type `finset ι`